mrp_sale_fields/models/mrp_production.py

```diff
@@ -20,10 +20,9 @@ def _compute_sale_order(self):
             (tuple(self.ids),),
         )
         production_data = self._cr.fetchall()
-        if production_data:
-            mapped_data = dict(production_data)
-            for production in self:
-                production.sale_id = mapped_data.get(production.id, False)
+        mapped_data = dict(production_data)
+        for production in self:
+            production.sale_id = mapped_data.get(production.id, False)
```

Comment:
As per discussed, since production_data will be an empty list if it's no data, `if production_data:` condition is removed.